packages/contracts-bedrock/src/L1/OPStackManager.sol

```diff
@@ -64,4 +340,186 @@ contract OPStackManager is ISemver {
         bytes19 first19Bytes = bytes19(hashedChainId);
         return address(uint160(bytes20(bytes.concat(versionByte, first19Bytes))));
     }
+
+    /// @notice Deterministically deploys a new proxy contract owned by the provided ProxyAdmin.
+    /// The salt is computed as a function of the L2 chain ID and the contract name. This is required
+    /// because we deploy many identical proxies, so they each require a unique salt for determinism.
+    function deployProxy(
+        uint256 _l2ChainId,
+        ProxyAdmin _proxyAdmin,
+        bytes32 _contractName
```

Comment:
This should be a string. I originally used bytes32 because its a smaller type (it can live on the stack, strings must be in memory or calldata), but feedback from the spec review was just use strings since it's simpler

The enum approach might be good too. Will check out the PR more closely later, but for now I'm not too worried about it since we should have tests that fail if we typo a string. But I do think you're right it's probably the way to go since type safety is always better

Reply:
Pushed the string change in 4717a7c82, let's revisit the enum approach in your PR